x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/engine/engine_search_preview/engine_search_preview.tsx

```diff
@@ -0,0 +1,230 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+
+import { useValues } from 'kea';
+
+import {
+  EuiBadge,
+  EuiBasicTable,
+  EuiBasicTableColumn,
+  EuiButton,
+  EuiFieldSearch,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiForm,
+  EuiFormRow,
+  EuiPanel,
+  EuiSelect,
+  EuiSpacer,
+} from '@elastic/eui';
+import { SearchProvider, SearchBox, Results, Sorting } from '@elastic/react-search-ui';
+import {
+  ResultsViewProps,
+  ResultViewProps,
+  InputViewProps,
+  SortingViewProps,
+} from '@elastic/react-search-ui-views';
+import { SearchDriverOptions, SearchResult } from '@elastic/search-ui';
+import EnginesAPIConnector, {
+  Transporter,
+  SearchReqest,
+  SearchResponse,
+} from '@elastic/search-ui-engines-connector';
+import { HttpSetup } from '@kbn/core-http-browser';
+import { i18n } from '@kbn/i18n';
+
+import { HttpLogic } from '../../../../shared/http';
+import { EngineViewTabs } from '../../../routes';
+import { EnterpriseSearchEnginesPageTemplate } from '../../layout/engines_page_template';
+
+import { EngineIndicesLogic } from '../engine_indices_logic';
+import { EngineViewLogic } from '../engine_view_logic';
+
+import { EngineSearchPreviewLogic } from './engine_search_preview_logic';
+
+class InternalEngineTransporter implements Transporter {
+  constructor(private http: HttpSetup, private engineName: string) {}
+
+  async performRequest(request: SearchReqest) {
+    const url = `/internal/enterprise_search/engines/${this.engineName}/search`;
+
+    const response = await this.http.post<SearchResponse>(url, {
+      body: JSON.stringify(request),
+    });
+
+    const withUniqueIds = {
```

Comment:
Yeh I hit this issue aswell elsewhere. Using the index name is a good idea and had similar ideas. Im certain this should be in Search UI but not yet sure of what the solution is. Lets talk about this more.